File: r/src/add_key.R
library(httr2)
library(jsonlite)
library(dotenv)
library(glue)

# Load environment variables
load_dot_env()

BASE_URL <- "https://api.hyperswitchai.com"

# Source the cache_token.R file to get access to its functions
source("src/cache_token.R")

main <- function() {
  tryCatch({
    # Get cached token using the imported function
    token <- get_token()  # Direct call to the function now

    # Add a new API key
    response <- request(glue("{BASE_URL}/admin/keys/add-key")) %>%
      req_headers(
        "Content-Type" = "application/json",
        "Authorization" = glue("Bearer {token}")
      ) %>%
      req_body_json(list(
        keyId = "claude-key-1",
        encryptionFragment = "OpenThePodBayDoorsHAL-2001",
        provider = "anthropic",
        apiKey = "sk-ant-api03-LhOEBAVFlPnTIy8e80m9lNkJpKBSV3v7xSwXkpX4f02lICRAAAAAAAAAAAAAAA"
      )) %>%
      req_perform()

    # Handle response
    if (resp_status(response) != 200) {
      if (resp_status(response) == 409) {
        error_json <- resp_body_json(response)
        if (!is.null(error_json$error) && error_json$error$code == "duplicate_key_id") {
          cat("\n\033[33m⚠️  Key already exists with this ID\033[0m\n")
          return()
        }
      }
      
      error_text <- resp_body_text(response)
      cat("\n\033[31mServer response:", resp_status(response), error_text, "\033[0m\n")
      stop(glue("Failed to add key: {resp_status(response)}"))
    }

    cat("\n\033[32m✅ Key added successfully\033[0m\n")

  }, error = function(e) {
    cat("Error:", e$message, "\n")
  })
}

# Execute if run directly
if (sys.nframe() == 0) {
  main()
} 

File: ruby/src/add_key.rb
require 'dotenv'
require 'faraday'
require 'json'

# Load environment variables from .env file
Dotenv.load

BASE_URL = 'https://api.hyperswitchai.com'

# Source the cache_token.rb file to get access to its functions
require_relative 'cache_token'

def main
  begin
    # Get cached token using the imported function
    token = get_token

    # Create HTTP client
    conn = Faraday.new(url: BASE_URL) do |f|
      f.request :json
      f.response :json
    end

    # Add a new API key
    response = conn.post('/admin/keys/add-key') do |req|
      req.headers['Content-Type'] = 'application/json'
      req.headers['Authorization'] = "Bearer #{token}"
      req.body = {
        keyId: 'claude-key-1',
        encryptionFragment: 'OpenThePodBayDoorsHAL-2001',
        provider: 'anthropic',
        apiKey: 'sk-ant-api03-LhOEBAVFlPnTIy8e80m9lNkJpKBSV3v7xSwXkpX4f02lICRAAAAAAAAAAAAAAA'
      }
    end

    # Handle response
    unless response.success?
      if response.status == 409
        error = response.body  # Body is already parsed as JSON
        if error.dig('error', 'code') == 'duplicate_key_id'
          puts "\n\e[33m⚠️  Key already exists with this ID\e[0m"
          return
        end
      end
      
      puts "\n\e[31mServer response: #{response.status} #{response.body}\e[0m"
      raise "Failed to add key: #{response.reason_phrase}"
    end

    puts "\n\e[32m✅ Key added successfully\e[0m"

  rescue => e
    puts "Error: #{e.message}"
  end
end

# Run if this file is executed directly
main if __FILE__ == $PROGRAM_NAME 